bagitobjecttransfer/caais/models.py

```diff
@@ -218,7 +218,8 @@ class Meta:
 
     @property
     def accession_identifier(self) -> str:
-        if self.identifiers:
+        # We check self.pk here in case the model is not saved yet
+        if self.pk and self.identifiers.count() > 0:
             accession_id = self.identifiers.accession_identifier()
```

Comment:
I understand that a reverse relationship is being used to get all the `Identifier` models are related to this instance of `Metadata`. However, I could not find the method `accession_identifier()` defined in the `Identifier` model class, so I'm not sure how `self.identifiers.accession_identifier()` is being called.

Reply:
Great question! If `metadata` is an instance of `caais.models.Metadata`, then `metadata.identifiers` is an instance of `caais.managers.IdentifierManager`. The identifiers property is actually a manager rather than a queryset of Identifier objects. The manager for the model is set here:

https://github.com/NationalCentreTruthReconciliation/Secure-Record-Transfer/blob/86ecc3cdb5bf4524bfb65733ac5583484b76993e/bagitobjecttransfer/caais/models.py#L466

And you can see that `accession_identifier()` is indeed a method defined on that manager:

https://github.com/NationalCentreTruthReconciliation/Secure-Record-Transfer/blob/86ecc3cdb5bf4524bfb65733ac5583484b76993e/bagitobjecttransfer/caais/managers.py#L55

Reply:
I see, thanks for clarifying!

Reply:
I see, thanks for clarifying!